### PR TITLE
refactor: replace di framework

### PR DESCRIPTION
**Pull request by bojiang**
## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature and improvements (non-breaking change which adds/improves functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Code Refactoring (internal change which is not user facing)
- [ ] Documentation
- [ ] Test, CI, or build

## Component(s) if applicable
- [ ] BentoService (service definition, dependency management, API input/output adapters)
- [ ] Model Artifact (model serialization, multi-framework support)
- [ ] Model Server (mico-batching, dockerisation, logging, OpenAPI, instruments)
- [ ] YataiService gRPC server (model registry, cloud deployment automation)
- [ ] YataiService web server (nodejs HTTP server and web UI)
- [ ] Internal (BentoML's own configuration, logging, utility, exception handling)
- [ ] BentoML CLI

## Checklist:

- [ ] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Comment by bojiang** [comment]
```
bentoml/configuration/__init__.py:139:4: W0611: Unused marshal imported from bentoml (unused-import)
117
bentoml/configuration/__init__.py:139:4: W0611: Unused server imported from bentoml (unused-import)
118
bentoml/configuration/__init__.py:139:4: W0611: Unused tracing imported from bentoml (unused-import)
119
bentoml/configuration/__init__.py:139:4: W0611: Unused cli imported from bentoml (unused-import)
120
bentoml/configuration/__init__.py:139:4: W0611: Unused adapters imported from bentoml (unused-import)
121
bentoml/configuration/__init__.py:139:4: W0611: Unused saved_bundle imported from bentoml (unused-import)
122
bentoml/configuration/__init__.py:139:4: W0611: Unused service imported from bentoml (unused-import)
123
bentoml/configuration/__init__.py:148:4: W0611: Unused yatai_service imported from bentoml.yatai (unused-import)
124
bentoml/configuration/__init__.py:149:4: W0611: Unused yatai_service_impl imported from bentoml.yatai (unused-import)
125
bentoml/configuration/__init__.py:150:4: W0611: Unused s3_repository imported from bentoml.yatai.repository (unused-import)
126
bentoml/configuration/__init__.py:150:4: W0611: Unused gcs_repository imported from bentoml.yatai.repository (unused-import)
```
We may avoid all these imports now.

File: bentoml/adapters/utils.py [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import gzip
 import json
 import os
-from dependency_injector.wiring import Provide, inject
+
+from simple_di import Provide, inject
```
Comment (parano) [inline bentoml/adapters/utils.py:5]: where is simple_di defined?

Comment (bojiang) [reply bentoml/adapters/utils.py:5]: setup.py

File: bentoml/tracing/__init__.py [hunk 2]
```diff
@@ -27,23 +28,27 @@
...
+    ],
+    jaeger_server_address: Optional[str] = Provide[
         BentoMLContainer.config.tracing.jaeger.address
     ],
-    jaeger_server_port: str = Provide[BentoMLContainer.config.tracing.jaeger.port],
+    jaeger_server_port: Optional[str] = Provide[
+        BentoMLContainer.config.tracing.jaeger.port
+    ],
 ):
     # isinstance check here allow trace to be used where the top-level entry point has
```
Comment (parano) [inline bentoml/tracing/__init__.py:41]: I guess we don't need this part anymore?

Comment (bojiang) [reply bentoml/tracing/__init__.py:41]: Yeah. Here is not the only place that we do not need anymore after introducing simple-di. I'll have a full cleaning after this PR.

File: tests/configuration/test_containers.py [hunk 3]
```diff
@@ -36,11 +37,12 @@ def test_override_none_value():
...
-    config = BentoMLConfiguration()
-    with pytest.raises(BentoMLConfigException) as e:
-        config.override(None, 6000)
-    assert e is not None
+# TODO: ?
+# def test_override_none_key():
+# config = BentoMLConfiguration()
+# with pytest.raises(BentoMLConfigException) as e:
+# config.override(None, 6000)
+# assert e is not None
```
Comment (ssheng) [inline tests/configuration/test_containers.py:45]: `override` related tests can be deleted.

**Review by ssheng — APPROVED** [review]

File: bentoml/configuration/containers.py [hunk 4]
```diff
@@ -277,3 +278,6 @@ class BentoMLContainer(containers.DeclarativeContainer):
         ),
         config.yatai.logging.path,
     )
+
+
+BentoMLContainer = BentoMLContainerClass()
```
Comment (parano) [inline bentoml/configuration/containers.py:283]: is this naming convention coming from the other dependency_injector library?
any reason not to just do `default_container = BentoMLContainer()`

Comment (parano) [reply bentoml/configuration/containers.py:283]: discussed offline, we will make the following rename in a follow up PR to reduce the changes here:

```
BentoMLContainerClass => BentoMLContainer
BentoMLContainer => BENTOML_CONTAINER
```

**Pull request merged**